src/smart-wallet.ts

```diff
@@ -0,0 +1,84 @@
+import { createSmartAccountClient } from 'permissionless';
+import { toSafeSmartAccount } from 'permissionless/accounts';
+import { createPimlicoClient } from 'permissionless/clients/pimlico';
+import type { Chain, Hex } from 'viem';
+import { createPublicClient, http } from 'viem';
+import { entryPoint07Address } from 'viem/account-abstraction';
+import { privateKeyToAccount } from 'viem/accounts';
+
+const DEFAULT_RPC_URL = 'https://rpc-geo-genesis-h0q2s21xx8.t.conduit.xyz';
+
+type GetSmartAccountWalletClientParams = {
+  privateKey: Hex;
+  pimlicoApiKey: string;
+  rpcUrl?: string;
+};
+
+export const getSmartAccountWalletClient = async ({
+  privateKey,
+  pimlicoApiKey,
+  rpcUrl: rpcUrlInput,
+  // biome-ignore lint/suspicious/noExplicitAny: we want to return the SmartAccountClient type, which is currently broken
+}: GetSmartAccountWalletClientParams): Promise<any> => {
```

Comment:
@baiirun super annoying to return `any` here, but otherwise `client.sendTransaction` type is wrong and requires an incorrect `account` argument. This is a TODO in their codebase

Reply:
What's the type if you don't set an explicit return type? Still `Promise<any>`? In the web app codebase it returns an actual object.

Reply:
You get an error:

```
The inferred type of 'getSmartAccountWalletClient' cannot be named without a reference to '~/node_modules/viem/_types/account-abstraction/types/userOperation.js'. This is likely not portable. A type annotation is necessary.
```